addons/pinniped/config-controller/controllers/utils_test.go

```diff
@@ -116,12 +116,19 @@ func verifySecretFunc(ctx context.Context, cluster *clusterapiv1beta1.Cluster, c
 		}
 
 		err = k8sClient.Get(ctx, client.ObjectKeyFromObject(secret), secret)
-		g.Expect(err).NotTo(HaveOccurred())
+
+		// if deleted, we expect an isnotfound error for v1alpha1
```

Comment:
Long term, I'd proabably prefer 2 separate testing funcs (or more) rather than 1 big func.  I think it reduces risk if we can edit 1 test and trust all other tests work as expected, but you can't really do that if changing tests requires touching "the brain". 

We can at least eliminate some of the `if` statements if we split out the v1alpha1 and v1alpha3 logic (though I do realize this causes some duplication, i think thats ok in test code). 

Happy to discuss!